mopidy/core/playback.py

```diff
@@ -534,6 +537,8 @@ def _trigger_track_playback_ended(self, time_position_before_stop):
 
     def _trigger_playback_state_changed(self, old_state, new_state):
         logger.debug("Triggering playback state change event")
+        if new_state == PlaybackState.STOPPED:
+            self._trigger_track_playback_ended(self.get_time_position())
```

Comment:
`_trigger_playback_state_changed()` is only called from `change_state()`. I'd rather have these two lines there, and keep the `_trigger...` functions clean of any "logic", only doing the `listener.CoreListener` call.

If I remove this, all tests still pass.

Reply:
Again, see comment above. Anything that looks odd is there because real world testing proved it was necessary.

Reply:
As for splitting it in two, I'm not sure what that would accomplish. The bug isn't fixed simply by the changes in tracklist.py. If you include just those you'll end up provoking race conditions and probably making it worse. Like I say, with race conditions that are so timing dependant, CI tests in a mock environment are of very limited use.